### PR TITLE
chore: update deps for go 1.20

### PR DESCRIPTION
**Pull request by DanStough**

### CODE REVIEW

File: mapping.go [hunk 1]
```diff
@@ -15,9 +15,9 @@ type assignmentKind interface {
 // singleAssignmentKind is a mapping operation between two fields that
 // ultimately are:
 //
-//  - basic
-//  - named structs
-//  - pointers to either of the above
+//   - basic
+//   - named structs
+//   - pointers to either of the above
```
Comment (DanStough) [inline mapping.go:20]: `go fmt` complaining

File: .circleci/config.yml [hunk 2]
```diff
@@ -5,11 +5,11 @@ workflows:
     jobs:
       - lint
       - go-test:
-          name: "test on go 1.17"
-          version: "1.17"
+          name: "test on go 1.19"
+          version: "1.20"
```
Comment (rboyer) [inline .circleci/config.yml:9]:
```suggestion
          version: "1.19"
```

File: go.mod [hunk 3]
```diff
@@ -4,10 +4,10 @@ go 1.14
 
 require (
```
Comment (rboyer) [inline go.mod:5]: It probably wouldn't hurt to bump the `go 1.14` directive as well in this file.

File: go.mod [hunk 4]
```diff
@@ -1,13 +1,20 @@
...
 )
+
+require (
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/pkg/errors v0.8.1 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	golang.org/x/mod v0.8.0 // indirect
+	golang.org/x/sys v0.5.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
```
Comment (DanStough) [inline go.mod:20]: I ran a tidy on this if you notice more go.* changes.

**Review by rboyer — APPROVED** [review]

**Pull request merged**